schemas/metaschema/DatabaseField.schema

```diff
@@ -16,35 +17,49 @@
   comment: { domain: 'Text' },
 
   Check: Validate(data => {
+    let kind = data.constructor.name;
+    if (kind === 'Function') return true;
+
     const isDomain = !!data.domain;
     const isCategory = !!data.category;
 
     // Required just one of fields: domain or category
-    if (isDomain && isCategory) return false;
-    if (!isDomain && !isCategory) return false;
+    if (
+      isDomain === isCategory && kind !== 'Index' && kind !== 'Unique'
+    ) {
+      return false;
+    }
 
     const fields = Object.keys(data);
 
     // Check allowed fields
     const allow = allowed => fields.every(f => allowed.includes(f));
 
-    let kind = data.constructor.name;
-    if (!kind && isCategory) kind = 'Category';
+    if (kind === 'Object' && isCategory) kind = 'Category';
     if (data.expression) kind = 'Expression';
 
     const allowedFields = {
-      Include: ['category', 'comment'],
+      Index: ['fields'],
+      Unique: ['fields'],
+      Include: ['category', 'comment', 'required'],
       Master: ['category', 'control', 'comment'],
       Many: ['category', 'comment'],
       Category: ['category', 'lookup', 'required', 'control', 'comment'],
-      Expression: ['domain', 'lookup', 'category', 'control', 'comment']
+      Expression: ['domain', 'lookup', 'category', 'control', 'comment'],
+      Object: [
+        'domain', 'required', 'unique', 'hidden', 'readOnly', 'default',
+        'example', 'control', 'validate', 'normalize', 'comment','index',
+      ]
     };
 
     const allowed = allowedFields[kind];
-    const fits = allowFields(allowed);
+    const fits = allow(allowed);
+    if (!fits) console.log(kind, data);
     if (!fits) return false;
 
     // Check index definition
-    return +data.index + +data.unique < 2;
+    return (data.index && !data.unique) ||
+        (data.unique && !data.index) ||
+        !(data.index && data.unique);
```

Comment:
`return data.index !== data.unique || !data.index`
or to be explicit
`return data.index !== data.unique || !data.index && !data.unique`
IIUC